### PR TITLE
cluster-api-provider-openstack: Update OWNERS

### PR DESCRIPTION
**Pull request by mdbooth**
See https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/OWNERS_ALIASES

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-openstack/OWNERS [hunk 1]
```diff
@@ -1,20 +1,10 @@
...
-- flaper87
-- Lion-Wei
-- chaosaffe
-- m1093782566
-- dims
-- sbueringer
-- jichenjc
-- hidekazuna
+  - sig-cluster-lifecycle-leads
+  - cluster-api-admins
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply config/jobs/kubernetes-sigs/cluster-api-provider-openstack/OWNERS:5]: Ah one moment

[the remark this answers is not in the document]
Comment (sbueringer) [reply config/jobs/kubernetes-sigs/cluster-api-provider-openstack/OWNERS:5]: It is used in CAPN and CAPV. I'll check what we want to do with that group

[the remark this answers is not in the document]
Comment (sbueringer) [reply config/jobs/kubernetes-sigs/cluster-api-provider-openstack/OWNERS:5]: @mdbooth The advice about inlining still applies. Can you please remove the cluster-api-admins alias? I will remove it in a separate PR

**Comment by sbueringer** [comment]
I can still approve this! :)
/approve
/lgtm

/hold
Feel free to merge when appropriate

**Comment by mdbooth** [comment]
This is purely mechanical so I can't see any reason to delay (other than wanting to leave @sbueringer in maintainers 😂).

/hold cancel

**Pull request merged**